test/hotspot/jtreg/compiler/c2/irTests/constantFold/TestOptional.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
```

Comment:
We would like to move all tests away from `irTests`. It would be better to sort tests by topic, rather than by the method we test them.